pkg/machine/e2e/machine_test.go

```diff
@@ -169,3 +172,16 @@ func copySparse(dst io.WriteSeeker, src io.Reader) (retErr error) {
 	_, err := io.Copy(spWriter, src)
 	return err
 }
+
+func getTmpDir(value string) string {
+	if runtime.GOOS == "darwin" {
+		if len(value) >= 22 || value == "" {
```

Comment:
I think I would still prefer that we only change the tmpdir if it is not set. If the user/runner of tests sets a path len > 22, I think we should fail with an error, and not secretly change the path without notice. 

macOS tests are passing now!